Error when building

When I try to build with
```
cargo build
```
It gives error:
```
me @ ~/code/src/paritytech/fleetwood - [master] $ cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading parity-hash v1.2.2                                                                                               
 Downloading bigint v4.4.1                                                                                                    
 Downloading pwasm-abi v0.1.10                                                                                                
 Downloading uint v0.3.0                                                                                                      
 Downloading pwasm-std v0.10.0                                                                                                
 Downloading pwasm-alloc v0.4.0                                                                                               
   Compiling proc-macro2 v0.4.20                                                                                              
   Compiling unicode-xid v0.1.0
   Compiling crunchy v0.1.6
   Compiling void v1.0.2
   Compiling wee_alloc v0.4.2
   Compiling serde v1.0.79
   Compiling libc v0.2.43
   Compiling memory_units v0.4.0
   Compiling byteorder v1.2.6
   Compiling cfg-if v0.1.5
   Compiling rustc-hex v2.0.1
   Compiling version_check v0.1.5
   Compiling rlibc v1.0.0
   Compiling either v1.5.0
   Compiling metrohash v1.0.5
   Compiling unreachable v1.0.0
   Compiling pwasm-libc v0.2.0
   Compiling lazy_static v1.1.0
   Compiling uint v0.3.0
   Compiling tiny-keccak v1.4.2
error[E0554]: #![feature] may not be used on the stable release channel
   --> /Users/me/.cargo/registry/src/github.com-1ecc6299db9ec823/wee_alloc-0.4.2/src/lib.rs:228:1
    |
228 | #![feature(alloc, allocator_api, core_intrinsics)]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0554`.
error: Could not compile `wee_alloc`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

System:
`cargo --version` cargo 1.29.0 (524a578d7 2018-08-05)
`rustc --version` rustc 1.29.1 (b801ae664 2018-09-20)

@Robbepop thanks, well it compiled more than before, so that's progress! The new errors are as follows:
```
scon @ ~/code/src/paritytech/fleetwood - [master] $ cargo +nightly build
   Compiling proc-macro2 v0.4.20                                                                                              
   Compiling unicode-xid v0.1.0                                                                                               
   Compiling crunchy v0.1.6                                                                                                   
   Compiling void v1.0.2                                                                                                      
   Compiling serde v1.0.79                                                                                                    
   Compiling wee_alloc v0.4.2                                                                                                 
   Compiling byteorder v1.2.6                                                                                                 
   Compiling memory_units v0.4.0                                                                                              
   Compiling rustc-hex v2.0.1                                                                                                 
   Compiling cfg-if v0.1.5                                                                                                    
   Compiling libc v0.2.43                                                                                                     
   Compiling version_check v0.1.5                                                                                             
   Compiling rlibc v1.0.0                                                                                                     
   Compiling metrohash v1.0.5                                                                                                 
   Compiling either v1.5.0                                                                                                    
   Compiling unreachable v1.0.0                                                                                               
   Compiling pwasm-libc v0.2.0                                                                                                
   Compiling lazy_static v1.1.0                                                                                               
   Compiling uint v0.3.0                                                                                                      
   Compiling tiny-keccak v1.4.2                                                                                               
   Compiling pwasm-alloc v0.4.0                                                                                               
   Compiling quote v0.6.8                                                                                                     
   Compiling syn v0.15.9                                                                                                      
   Compiling serde_derive v1.0.79                                                                                             
   Compiling bincode v1.0.1                                                                                                   
   Compiling parity-hash v1.2.2                                                                                               
   Compiling bigint v4.4.1                                                                                                    
   Compiling pwasm-std v0.10.0                                                                                                
   Compiling pwasm-abi v0.1.10                                                                                                
   Compiling fleetwood v0.1.0 (/Users/me/code/src/paritytech/fleetwood)                                                     
error[E0277]: the trait bound `bigint::U256: pwasm_abi::eth::AbiType` is not satisfied                                        
   --> src/lib.rs:489:32                                                                                                      
    |                                                                                                                         
489 |                           stream.pop()                                                                                  
    |                                  ^^^ the trait `pwasm_abi::eth::AbiType` is not implemented for `bigint::U256`          
...                                                                                                                           
524 | / impl_solidityargs_abitype! {                                                                                          
525 | |     Vec<u8>,                                                                                                          
526 | |     U256,                                                                                                             
527 | |     ::parity_hash::Address,                                                                                           
...   |                                                                                                                       
565 | |     [u8; 32],                                                                                                         
566 | | }                                                                                                                     
    | |_- in this macro invocation                                                                                            
                                                                                                                              
error[E0277]: the trait bound `bigint::U256: pwasm_abi::eth::AbiType` is not satisfied                                        
   --> src/lib.rs:495:32                                                                                                      
    |                                                                                                                         
495 |                           stream.pop()                                                                                  
    |                                  ^^^ the trait `pwasm_abi::eth::AbiType` is not implemented for `bigint::U256`          
...                                                                                                                           
524 | / impl_solidityargs_abitype! {                                                                                          
525 | |     Vec<u8>,                                                                                                          
526 | |     U256,                                                                                                             
527 | |     ::parity_hash::Address,                                                                                           
...   |                                                                                                                       
565 | |     [u8; 32],                                                                                                         
566 | | }                                                                                                                     
    | |_- in this macro invocation                                                                                            
    |                                                                                                                         
    = note: required because of the requirements on the impl of `pwasm_abi::eth::AbiType` for `std::vec::Vec<bigint::U256>`   
                                                                                                                              
error[E0277]: the trait bound `bigint::U256: pwasm_abi::eth::AbiType` is not satisfied                                        
   --> src/lib.rs:501:30                                                                                                      
    |                                                                                                                         
501 |                           sink.push(self)                                                                               
    |                                ^^^^ the trait `pwasm_abi::eth::AbiType` is not implemented for `bigint::U256`           
...                                                                                                                           
524 | / impl_solidityargs_abitype! {                                                                                          
525 | |     Vec<u8>,                                                                                                          
526 | |     U256,                                                                                                             
527 | |     ::parity_hash::Address,                                                                                           
...   |                                                                                                                       
565 | |     [u8; 32],                                                                                                         
566 | | }                                                                                                                     
    | |_- in this macro invocation                                                                                            
                                                                                                                              
error[E0277]: the trait bound `bigint::U256: pwasm_abi::eth::AbiType` is not satisfied                                        
   --> src/lib.rs:513:30                                                                                                      
    |                                                                                                                         
513 |                           sink.push(self)                                                                               
    |                                ^^^^ the trait `pwasm_abi::eth::AbiType` is not implemented for `bigint::U256`           
...                                                                                                                           
524 | / impl_solidityargs_abitype! {                                                                                          
525 | |     Vec<u8>,                                                                                                          
526 | |     U256,                                                                                                             
527 | |     ::parity_hash::Address,                                                                                           
...   |                                                                                                                       
565 | |     [u8; 32],                                                                                                         
566 | | }                                                                                                                     
    | |_- in this macro invocation                                                                                            
    |                                                                                                                         
    = note: required because of the requirements on the impl of `pwasm_abi::eth::AbiType` for `std::vec::Vec<bigint::U256>`   
                                                                                                                              
error: aborting due to 4 previous errors                                                                                      
                                                                                                                              
For more information about this error, try `rustc --explain E0277`.                                                           
error: Could not compile `fleetwood`.                                                                                         

To learn more, run the command again with --verbose.

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,9 +11,8 @@ either = "1.5"
 tiny-keccak = "1.4"
 bincode = "1.0"
 metrohash = "1.0"
-parity-hash = { version = "1.2", features = ["serialize"] }
-bigint = { version = "4.3.0", features = ["serialize"] }
-pwasm-abi = { version = "0.1", features = ["std"] }
+pwasm-abi = { version = "0.2", features = ["std"] }
+pwasm-std = { version = "0.13" }
 
 [features]
 default = []

--- a/src/dummy.rs
+++ b/src/dummy.rs
@@ -1,4 +1,4 @@
-use parity_hash::{Address, H256};
+use pwasm_std::types::{Address, H256};
 use std::collections::HashMap;
 use std::sync::Mutex;
 
@@ -19,14 +19,14 @@ pub fn origin() -> Address {
 }
 
 pub fn write(key: &H256, val: &[u8; 32]) {
-    STORAGE_MUTEX.lock().unwrap().insert(key.0, val.clone());
+    STORAGE_MUTEX.lock().unwrap().insert(key.to_fixed_bytes(), val.clone());
 }
 
 pub fn read(key: &H256) -> [u8; 32] {
     STORAGE_MUTEX
         .lock()
         .unwrap()
-        .get(&key.0)
+        .get(key.as_bytes())
         .cloned()
         .unwrap_or([0; 32])
 }

--- a/src/environment.rs
+++ b/src/environment.rs
@@ -1,4 +1,4 @@
-use parity_hash::H256;
+use pwasm_std::types::H256;
 use std::io::{self, Cursor};
 
 pub enum Writeable<'a> {
@@ -80,7 +80,7 @@ impl Key for H256 {
         out[5] = (val << 16) as u8;
         out[6] = (val << 8) as u8;
         out[7] = val as u8;
-        H256(out)
+        H256::from(out)
     }
 }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -23,20 +23,21 @@ extern crate lazy_static;
 #[macro_use]
 extern crate serde_derive;
 
-extern crate bigint;
 extern crate bincode;
 extern crate either;
 extern crate metrohash;
-extern crate parity_hash;
+extern crate pwasm_std;
 extern crate pwasm_abi;
 extern crate serde;
 extern crate tiny_keccak;
 
-use bigint::U256;
 use environment::{Key, Value};
 use metrohash::MetroHash;
-use parity_hash::{Address, H256};
-use pwasm_abi::eth::{Sink, Stream};
+use pwasm_std::types::{Address, H256};
+use pwasm_abi::{
+    eth::{Sink, Stream},
+    types::U256,
+};
 use serde::{Deserialize, Serialize};
 use std::any::Any;
 use std::cell::{Cell, UnsafeCell};
@@ -524,8 +525,8 @@ macro_rules! impl_solidityargs_abitype {
 impl_solidityargs_abitype! {
     Vec<u8>,
     U256,
-    ::parity_hash::Address,
-    ::parity_hash::H256,
+    pwasm_std::types::Address,
+    pwasm_std::types::H256,
     u32,
     u64,
     i32,
@@ -874,7 +875,7 @@ impl_soltype!(i16, "int16");
 impl_soltype!(i32, "int32");
 impl_soltype!(i64, "int64");
 impl_soltype!(U256, "uint256");
-impl_soltype!(::parity_hash::Address, "address");
+impl_soltype!(pwasm_std::types::Address, "address");
 
 macro_rules! sol_array {
     (@capture $e:expr) => {
@@ -1336,8 +1337,8 @@ mod test {
             [(i >> 24) as u8, (i >> 16) as u8, (i >> 8) as u8, i as u8]
         }
 
-        use bigint::U256;
-        use parity_hash::Address;
+        use pwasm_abi::types::U256;
+        use pwasm_std::types::Address;
 
         messages! {
             totalSupply() -> U256;
@@ -1417,8 +1418,8 @@ mod test {
 
     #[test]
     fn erc20() {
-        use bigint::U256;
-        use parity_hash::Address;
+        use pwasm_abi::types::U256;
+        use pwasm_std::types::Address;
         use {Contract, ContractDef, Database, DummyEnv};
 
         messages! {
@@ -1471,7 +1472,7 @@ mod test {
             });
 
         let total = U256::from(1_000_000u64);
-        let transfer_to = Address::from(&[123; 160][..]);
+        let transfer_to = Address::from([123; 160]);
         let mut contract = definition.deploy(&mut env, total.clone());
 
         let total_in_contract = contract.call::<TotalSupply>(());

